server/pulp/plugins/util/metadata_writer.py

```diff
@@ -124,7 +124,7 @@ def _open_metadata_file_handle(self):
             parent_dir = os.path.dirname(self.metadata_file_path)
 
             if not os.path.exists(parent_dir):
```

Comment:
you don't need to check for presence of the path here, remove the if condition

Reply:
The os.access() check in the elif is only needed if we're not creating the path in the mkdir() call. I tried a couple of iterations here, and decided that while this one is a little redundant around the mkdir(), it seemed more clear in terms of understanding the code. What do you think?

Reply:
oooh right, i did not notice the elif statement, i am cool with leaving if